### PR TITLE
Qr Code Page for Events

### LINKED ISSUE #236: Create QR Code Page that contains a QR Code component
As title. The QR Code should lead the user to an event's Inductee/Guest page (implemented separately from this issue) if the user is not logged in, and it should lead to an event's Event Details Page (for now) if the user is already logged in.

For the QR Code portion, there are plenty of libraries and packages that help with this. Some examples are [qrcode.react](https://www.npmjs.com/package/qrcode.react) and [react-qr-code](https://www.npmjs.com/package/react-qr-code). If you want something more general, as in not already made into a React component like the two packages above, use [node-qrcode](https://github.com/soldair/node-qrcode). Otherwise, use whatever you feel like would work.

For this to work as specified in the first paragraph, you might have to make the QR Code lead to an intermediate page that handles the logic of routing to and rendering which page (Inductee/Guest page or EventDetails page) based on whether the user is logged in or not (check out the event creation page for an example). If you find another way to do this that you think is better, go for it.

Put this QR Code component in its own page. Do not put this in the guest event sign in page. You also might want to add a title to the QR Code page. Perhaps the title could be the event name?

Refer to the HKN Software Development Planning doc for more information regarding the workflow. PM me @thaigillespie if you do not have access to the doc yet or forgot where it is at.

Update: As a suggestion, you might be able to check whether the user is logged in or not **before** generating the QR code. After the check, use the appropriate URL to generate the QR code. This is yet another way to approach this issue, other than the way listed above in the third paragraph.

### PR DESCRIPTION
**Pull request by Titan-Ngo**
Closes #236 

I followed the Qr Code Intermediate page suggestion, which directs users depending on whether they are signed in or not. Thus, two new pages were made: QrCodePage (accessed via a button from the EventDetails page) and QrCodeIntermediatePage (accessed via scanning the QR Code on the QrCodePage). It was unspecified what information to include on the QrCodePage, so I included most of the Event details on Event modals.

Below are videos showing the pages:

Signed in placeholder - Event details page.
[QrCode Usage - Signed In User](https://user-images.githubusercontent.com/42155527/129466027-35e6c7ee-ac13-4660-977d-c8f5c1f961fd.mp4
)

Not signed in placeholder - Empty page.
[QrCode Usage - Not Signed In](https://user-images.githubusercontent.com/42155527/129466041-1e8ff23f-1fb1-4792-9a9e-385359a98810.mp4)

QrCode Mobile Phone Scanning - Gives Correct URL:
[QrCode Scanning - Gives Correct URL](https://user-images.githubusercontent.com/42155527/129466112-6898b516-070e-44a0-957d-2e5b015431d0.mp4)

### CODE REVIEW

File: frontend/src/pages/QrCodeIntermediatePage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  return (
+    <UserContext.Consumer>
+      {userClaims => {
+        if (userClaims != null) {
+          const { userRoles } = userClaims;
+
+          if (userRoles != null) {
+            // user is signed in, route to event details page (placeholder)
+            history.push(ROUTES.EVENT_DETAILS_WITH_ID(eventId));
```
Comment (thaigillespie) [inline frontend/src/pages/QrCodeIntermediatePage/index.tsx:29]: I think this is fine, especially if QR code will only ever be used for events stuff (it is for now, who knows what will happen in the future).

**Review by thaigillespie — CHANGES_REQUESTED** [review]
Looks decent so far. I left some comments regarding your code for you to look at

File: frontend/src/pages/QrCodeIntermediatePage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          }
+        }
+
+        return (
+          <Grid container direction='column'>
```
[the remark this answers is not in the document]
Comment (audreygao) [reply frontend/src/pages/QrCodeIntermediatePage/index.tsx:34]: I was reading through #236, if the user is not logged in, I think it should redirect to the inductee/guest page I worked on? 'events/id/signinoptions'

[the remark this answers is not in the document]
Comment (thaigillespie) [reply frontend/src/pages/QrCodeIntermediatePage/index.tsx:34]: Yes, you're right. Good catch. Alright just do what Audrey said here @Titan-Ngo

**Comment by Titan-Ngo** [comment]
- redirected user to  'events/id/signinoptions' when not signed in 
- replaced usage of Location State to using eventId (passed as a route param) to get event info

**Review by thaigillespie — APPROVED** [review]
Lgtm. Good job!

**Review by audreygao — APPROVED** [review]
looks good to me

**Pull request merged**